src/app/activity/bounty/Claim.ts

```diff
@@ -68,56 +100,147 @@ const getDbHandler = async (request: ClaimRequest): Promise<{dbBountyResult: Bou
     const db: Db = await MongoDbUtils.connect('bountyboard');
-	const bountyCollection = db.collection('bounties');
+    const bountyCollection = db.collection('bounties');
+    let claimedBounty: BountyCollection;
+    const currentDate = (new Date()).toISOString();
+
+    // If claiming an evergreen bounty, create a copy and use that
+    if (dbBountyResult.evergreen) {
+        const bountyRec: BountyCollection = Object.assign({}, dbBountyResult);
+        bountyRec.parentId = bountyRec._id;
+        delete bountyRec._id;
+        delete bountyRec.isParent;
+        delete bountyRec.childrenIds;
+        delete bountyRec.claimLimit;
+        const claimedInsertResult = await bountyCollection.insertOne(bountyRec);
+        if (claimedInsertResult == null) {
+            Log.error('failed to create claimed bounty from evergreen');
+            throw new Error('Sorry something is not working, our devs are looking into it.');
+        }
+        claimedBounty = await bountyCollection.findOne({_id: claimedInsertResult.insertedId});
+        let updatedBountyResult: UpdateWriteOpResult = await bountyCollection.updateOne({ _id: new mongo.ObjectId(dbBountyResult._id) }, {
+            $push: {
+                childrenIds: claimedBounty._id
+            }
+        });
+        if (updatedBountyResult == null) {
+            Log.error('failed to update evergreen bounty with claimed Id');
+            throw new Error('Sorry something is not working, our devs are looking into it.');
+        }
 
-	const dbBountyResult: BountyCollection = await bountyCollection.findOne({
-		_id: new mongo.ObjectId(request.bountyId),
-		status: BountyStatus.open,
-	});
+        // Pull it back for second update
+        dbBountyResult = await bountyCollection.findOne({
+            _id: new mongo.ObjectId(dbBountyResult._id)
+        });
     
-    const currentDate = (new Date()).toISOString();
-	const writeResult: UpdateWriteOpResult = await bountyCollection.updateOne(dbBountyResult, {
-		$set: {
-			claimedBy: {
-				discordHandle: claimedByUser.user.tag,
-				discordId: claimedByUser.user.id,
-				iconUrl: claimedByUser.user.avatarURL(),
-			},
-			claimedAt: currentDate,
-			status: BountyStatus.in_progress,
-		},
-		$push: {
-			statusHistory: {
-				status: BountyStatus.in_progress,
-				setAt: currentDate,
-			},
-		},
-	});
+
+        // If we have hit the claim limit, close this bounty
+        if (dbBountyResult.claimLimit !== undefined) {
+            const claimedCount = (dbBountyResult.childrenIds !== undefined ? dbBountyResult.childrenIds.length : 0);
+            if (claimedCount >= dbBountyResult.claimLimit) {
+                updatedBountyResult = await bountyCollection.updateOne(dbBountyResult, {
+                    $set: {
+                        // TODO is leaving DeletedBy empty OK? Can assume deletion happened automatically in that case
```

Comment:
I think it's fine. Also feel free to use some application constant like `DELETED_BY_BOUNTYBOARD_APP`. `deletedBy` isn't type checked in BountyCollection, so AFAIK it won't break anything

Reply:
I guess I'd rather leave it. Could cause confusing issues in the future